### DISCUSSION TITLE
How to define schema for Map<String, String>

### DISCUSSION THREAD

**Discussion by venkatpselvam1** [json-schema-org/community#207] [Q&A] [answered]
I have a simple java dictionary as HashMap<String, String>. The sample json will look like the below format. The no of elements in the dict may be zero or more. The values are not complex object the are simple strings. How can I write a validator for this?

```
{
   "a": "a1",
   "b": "b1",
   "c": "c1",
   "d": "d1",
}
```

**Comment by gregsdennis** [comment, marked as answer]
The keyword you want is `additionalProperties`.

```json
{
  "type": "object",
  "additionalProperties": {
    "type": "string"
  }
}
```

This keyword will validate any properties that don't appear in `properties` or `patternProperties`.

**Comment by HACKERPRO456** [comment]
https://github.com/json-schema-org/community/discussions/207#discussioncomment-3219441
